From 55809dee03759fdae2cbdd0234ba3d103c8740b7 Mon Sep 17 00:00:00 2001
From: Jason Andryuk <jandryuk@gmail.com>
Date: Fri, 20 Sep 2019 10:53:45 -0400
Subject: [PATCH 08/26] flask: Add pv_disk & pv_net interfaces

domain_comms is a large interface that opens up bi-directional use of
grants.  Actual pv driver front back operations are asymmetric with the
frontend providing and the backend mapping the grants.

Supplement domain_comms with the smaller pv_net & pv_disk interaces.
They don't prevent a communications channel, but they help ensure
correctness of configuration by requiring correct use of frontend and
backends.

Signed-off-by: Jason Andryuk <jandryuk@gmail.com>
---
 tools/flask/policy/modules/xen.if | 15 +++++++++++++++
 1 file changed, 15 insertions(+)

diff --git a/tools/flask/policy/modules/xen.if b/tools/flask/policy/modules/xen.if
index db4a6dc892..962c3a7cb1 100644
--- a/tools/flask/policy/modules/xen.if
+++ b/tools/flask/policy/modules/xen.if
@@ -130,6 +130,21 @@ define(`domain_comms', `
 	allow $2 $1:grant { map_read map_write copy unmap };
 ')
 
+# pv_net(dom_fe, dom_be)
+#   Allow dom_fe to use a PV netback in dom_be
+define(`pv_net', `
+	domain_event_comms($1, $2)
+	allow $2 $1:grant { map_read map_write copy unmap };
+	allow $1 $2:grant { copy };
+')
+
+# pv_disk(dom_fe, dom_be)
+#   Allow dom_fe to use a PV blkback in dom_be
+define(`pv_disk', `
+	domain_event_comms($1, $2)
+	allow $2 $1:grant { map_read map_write unmap };
+')
+
 # domain_self_comms(domain)
 #   Allow a non-singleton domain type to communicate with itself using grants
 #   and event channels
-- 
2.21.0

